### PR TITLE
Initial addition of health check

### PR DESCRIPTION
**Pull request by kadaan**
First stab at implementing #348

### CODE REVIEW

**Comment by kadaan** [comment]
@joelgriffith @douglasmiranda I'm not entirely happy with how the semantics of these health checks are.  What is a good proxy to determine system health?

File: src/browserless.ts [hunk 1]
```diff
@@ -342,8 +359,6 @@ export class BrowserlessServer {
     await this.puppeteerProvider.close();
 
     debug(`Successfully shutdown, exiting`);
-
-    process.exit(0);
```
Comment (joelgriffith) [inline src/browserless.ts]: We still might need this to force the process to close (unless lightship does this somehow?)

**Comment by joelgriffith** [comment]
I wonder if we should loop in hardware monitoring into this? We can check those metrics (CPU/Memory) which are user-defined, to determine "health"

File: src/browserless.ts (hunk 1 above)
Comment (kadaan) [reply src/browserless.ts]: It does, AFAIK.  
> Do not call process.exit() in a shutdown handler – Lighthouse calls process.exit() after all registered shutdown handlers have run to completion.

**Comment by kadaan** [comment]
That might work.

Was also wondering if we should indicate health by "success" of calls going to the underlying chrome instances.  "success" being defined as a call getting run and completing, regardless of the http result.  Normally, this could be determined by the last X real calls (or real calls over X time).  In low volume circumstances, we could automatically submit canary calls that would run in chrome and complete very quickly with no outside dependencies.

**Pull request closed without merging**